test/core/currency/wallet/currency-wallet.test.js

```diff
@@ -195,25 +196,49 @@ describe('currency wallets', function () {
     expect(await wallet.nativeToDenomination('10', 'TOKEN')).equals('0.01')
   })
 
-  // it('can have metadata', function () {
-  //   const store = makeFakeCurrencyStore()
-  //
-  //   return makeFakeCurrencyWallet(store).then(wallet => {
-  //     const tx = { txid: 'a', metadata: { name: 'me' } }
-  //     store.dispatch({
-  //       type: 'SET_TXS',
-  //       payload: [{ txid: 'a', nativeAmount: '25' }]
-  //     })
-  //     return wallet.saveTx(tx).then(() =>
-  //       wallet.getTransactions({}).then(txs => {
-  //         assert.equal(txs.length, 1)
-  //         assert.strictEqual(txs[0].metadata.name, tx.metadata.name)
-  //         assert.strictEqual(txs[0].metadata.amountFiat, 0.75)
-  //         assert.strictEqual(txs[0].amountSatoshi, 25)
-  //         assert.strictEqual(txs[0].nativeAmount, '25')
-  //         return null
-  //       })
-  //     )
-  //   })
-  // })
+  it('can have metadata', async function () {
+    const [wallet, config] = await makeFakeCurrencyWallet()
+
+    const tx: any = { txid: 'a', metadata: { name: 'me' } }
+    await config.changeUserSettings({ txs: { a: { nativeAmount: '25' } } })
+    await wallet.saveTx(tx)
+
+    const txs = await wallet.getTransactions({})
+    expect(txs.length).equals(1)
+    expect(txs[0].nativeAmount).equals('25')
+
+    const { metadata } = tx
+    if (metadata == null) throw new Error('No metadata')
+    expect(metadata.name).equals(metadata.name)
```

Comment:
Shouldn't this be 

`expect(metadata.name).equals('me')`

Reply:
Fixed.